### PR TITLE
Enhance nist_kw with some NULL buffers tests

### LINKED ISSUE #1882: nist_kw: possible bug when a size argument is 0
According to https://github.com/ARMmbed/mbedtls/pull/1658#issuecomment-407013183 it seems that a test was failing and part of the fix involved the [following patch](https://gist.github.com/Patater/9fc367023e2ae68ccc9e94173e612724) (which has been erased from the history due to a post-review but pre-merge rebase):
```
diff --git a/tests/suites/test_suite_nist_kw.function b/tests/suites/test_suite_nist_kw.function
index 2724fbdb1..eb67c03f0 100644
--- a/tests/suites/test_suite_nist_kw.function
+++ b/tests/suites/test_suite_nist_kw.function
@@ -158,7 +158,16 @@ void nist_kw_plaintext_lengths( int in_len, int out_len, int mode, int res )
 
     memset( key, 0, sizeof( key ) );
 
-    plaintext = mbedtls_calloc( 1, in_len );
+    if (in_len == 0)
+    {
+        /* mbedtls_calloc can return NULL for zero-length buffers. Make sure we
+         * always have a plaintext buffer, even if the length is 0. */
+        plaintext = mbedtls_calloc( 1, 1 );
+    }
+    else
+    {
+        plaintext = mbedtls_calloc( 1, in_len );
+    }
     TEST_ASSERT( plaintext != NULL );
     ciphertext = mbedtls_calloc( 1, output_len );
     TEST_ASSERT( ciphertext != NULL );
```
This can't be right. `in_len == 0` and `plaintext == NULL` should be valid input. If something failed then this is a product issue, not a test issue.

### PR DESCRIPTION
**Pull request by RonEld**
## Description
Enhance the nist_kw test suite, with setting zero length input\output
buffers. Resolves #1882.
Increasing test coverage, by testing NULL pointers with zero length

## Status
**READY**

## Requires Backporting

 NO  

## Todos
- [x] Tests
- [ ] Documentation
- [ ] Changelog updated
- [ ] Backported

## Steps to test or reproduce
Outline the steps to test or reproduce the PR here.

### CODE REVIEW

File: tests/suites/test_suite_nist_kw.function [hunk 1]
```diff
@@ -158,19 +158,17 @@ void nist_kw_plaintext_lengths( int in_len, int out_len, int mode, int res )
...
-    if (in_len == 0)
+    if( in_len != 0 )
     {
-        /* mbedtls_calloc can return NULL for zero-length buffers. Make sure we
-         * always have a plaintext buffer, even if the length is 0. */
-        plaintext = mbedtls_calloc( 1, 1 );
+        plaintext = mbedtls_calloc( 1, in_len );
     }
-    else
+    TEST_ASSERT( in_len == 0 || plaintext != NULL );
```
Comment (gilles-peskine-arm) [inline tests/suites/test_suite_nist_kw.function]: Move this inside the `if( if_len != 0 )` and simplify it to `TEST_ASSERT( plaintext != NULL )`.

**Comment by RonEld** [comment]
@gilles-peskine-arm I addressed your comments
I also ran all.sh in my environment, to check there isn't regression, and no issues rose

File: tests/suites/test_suite_nist_kw.function [hunk 2]
```diff
@@ -161,14 +161,14 @@ void nist_kw_plaintext_lengths( int in_len, int out_len, int mode, int res )
...
     {
         plaintext = mbedtls_calloc( 1, in_len );
+        TEST_ASSERT( plaintext != NULL );
     }
-    TEST_ASSERT( in_len == 0 || plaintext != NULL );
 
     if( out_len != 0 )
     {
         ciphertext = mbedtls_calloc( 1, output_len );
+    TEST_ASSERT( ciphertext != NULL );
```
Comment (Patater) [inline tests/suites/test_suite_nist_kw.function]: Fix indentation here.

Comment (RonEld) [reply tests/suites/test_suite_nist_kw.function]: done

**Comment by RonEld** [comment]
@Patater I have fixed according to your comment. 
Please review

**Pull request merged**